bibtex-actions.el

```diff
@@ -0,0 +1,116 @@
+;;; bibtex-actions.el --- Bibliography manager action
+
+;;; Code:
+
+(require 'bibtex-completion)
+
+(defvar bibtex-actions-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map (kbd "p") 'bibtex-actions-open-pdf)
+    (define-key map (kbd "u") 'bibtex-actions-open-url-or-doi)
+    (define-key map (kbd "c") 'bibtex-actions-insert-citation)
+    (define-key map (kbd "r") 'bibtex-actions-insert-reference)
+    (define-key map (kbd "k") 'bibtex-actions-insert-key)
+    (define-key map (kbd "b") 'bibtex-actions-insert-bibtex)
+    (define-key map (kbd "a") 'bibtex-actions-add-PDF-attachment)
+    (define-key map (kbd "e") 'bibtex-actions-edit-notes)
+    (define-key map (kbd "s") 'bibtex-actions-show-entry)
+    (define-key map (kbd "l") 'bibtex-actions-add-pdf-to-library)
+    map))
+
+(defun bibtex-actions--get-candidates ()
+  "Return all keys from bibtex-completion-candidates."
+  (mapcar
+   (lambda (cand)
+     (cons (bibtex-completion-format-entry cand (1- (frame-width)))
+           (cdr (assoc "=key=" cand))))
+   (bibtex-completion-candidates)))
```

Comment:
This suggestion uses `cl-loop`, as elsewhere in the codebase, and for better performance, and also uses propertize to allow us to support `bibtex-completion-additiona-search-fields`.

```suggestion
(defun bibtex-completion--get-candidates ()
  "Return all keys from bibtex-completion-candidates."
  (cl-loop
   for candidate in (bibtex-completion-candidates)
   collect
   (cons
    ;; Here use one string for display, and the other for search.
    (propertize
     (car candidate) 'display (bibtex-completion-format-entry candidate (1- (frame-width))))
    (cdr (assoc "=key=" candidate)))))
```